homeassistant/components/webmin/sensor.py

```diff
@@ -0,0 +1,119 @@
+        native_unit_of_measurement=UnitOfInformation.KIBIBYTES,
+        suggested_unit_of_measurement=UnitOfInformation.GIBIBYTES,
+        device_class=SensorDeviceClass.DATA_SIZE,
+        suggested_display_precision=2,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key="mem_free",
+        name="Free Memory",
+        icon="mdi:memory",
+        native_unit_of_measurement=UnitOfInformation.KIBIBYTES,
+        suggested_unit_of_measurement=UnitOfInformation.GIBIBYTES,
+        device_class=SensorDeviceClass.DATA_SIZE,
+        suggested_display_precision=2,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key="swap_total",
+        name="Total Swap",
+        icon="mdi:memory",
+        native_unit_of_measurement=UnitOfInformation.KIBIBYTES,
+        suggested_unit_of_measurement=UnitOfInformation.GIBIBYTES,
+        device_class=SensorDeviceClass.DATA_SIZE,
+        suggested_display_precision=2,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key="swap_free",
+        name="Free Swap",
+        icon="mdi:memory",
+        native_unit_of_measurement=UnitOfInformation.KIBIBYTES,
+        suggested_unit_of_measurement=UnitOfInformation.GIBIBYTES,
+        device_class=SensorDeviceClass.DATA_SIZE,
+        suggested_display_precision=2,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up Webmin sensors based on a config entry."""
+    async_add_entities(
+        [
+            WebminSensor(hass.data[DOMAIN][entry.entry_id], description)
+            for description in SENSOR_TYPES
+        ]
+    )
+
+
+class WebminSensor(CoordinatorEntity[WebminUpdateCoordinator], SensorEntity):
+    """Represents a Webmin sensor."""
+
+    entity_description: SensorEntityDescription
+
+    def __init__(
+        self, coordinator: WebminUpdateCoordinator, description: SensorEntityDescription
+    ) -> None:
+        """Initialize a Webmin sensor."""
+
+        super().__init__(coordinator)
+        self.entity_description = description
+        self._attr_device_info = coordinator.device_info
+        self._attr_unique_id = f"{coordinator.mac_address}_{description.key}"
+        if self.entity_description.key in self.coordinator.data:
+            self._attr_available = True
+            self._attr_native_value = self.coordinator.data[self.entity_description.key]
+        else:
+            self._attr_available = False
+
+    def _handle_coordinator_update(self) -> None:
+        if self.entity_description.key in self.coordinator.data:
+            self._attr_available = True
+            self._attr_native_value = self.coordinator.data[self.entity_description.key]
+        else:
+            self._attr_available = False
+        self.async_write_ha_state()
```

Comment:
Why not make use of the `native_value` property function?

Besides, I would move the if to `async_setup_entry` and just setup entities if the description key is in the coordinator data. This way you wont end up with unavailable entities.

Also, the `CoordinatorEntity` takes care of `_attr_available` if a update fails

Reply:
Yes, you are right, thanks